File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        values = []
        
        def fill_levels(node, level):
          if node is None:
            return
          if len(values) == level:
            values.append(node.val)
          fill_levels(node.left, level + 1)
          fill_levels(node.right, level + 1)
          
        fill_levels(root, 0)
        return values[-1]

File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findBottomLeftValue = function(root) {
  let ans = 0;
  let height = 0;
  let queue = [root];
  // BFS
  while (queue.length) {
    const node = queue.shift();
    ans = node.val;
    if (node.right) queue.push(node.right);
    if (node.left) queue.push(node.left);
  }
  return ans;
};